### PR TITLE
Added test case for base theme SVN functionality

### PR DESCRIPTION
**Pull request by nwinkler**
Unit tests for #1612, #1613.

Want to take a look, @arturmartins, @davesteinberg?

### CODE REVIEW

**Comment by davesteinberg** [comment]
@nwinkler Is a `.svn` directory even needed in the setup? The big problem we were seeing is that `svn` was being run in _every_ directory (presumably,`svn` is being used to detect if it's a repo or not).

**Comment by nwinkler** [comment]
Yes, the `.svn` directory is necessary for the test. If you look at the code of the `scm` function, you will see the two branches of the code we care about for SVN:

https://github.com/Bash-it/bash-it/blob/ecd58894f74a84c5733dce139ed7af068d372181/themes/base.theme.bash#L100-L111

* Line 107 checks for the presence of the `.svn` directory and then uses SVN.
* Line 108 calls `svn info ...` which checks for a `.svn` folder in a parent folder.
* If none of these is true, then line 109 will set the SCM variable to `NONE`.

I've added some more test cases to cover both line 107 and 108 with a working and a broken `svn` command.

**Comment by davesteinberg** [comment]
@nwinkler Ah, I see. Thanks for the explanation!

File: test/fixtures/svn/working/svn [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+# If the info command is called
+# AND the parent folder contains the .svn folder
+# THEN return the current path, similar to what `svn info` does
+if [[ "$1" = "info" ]] && [[ -d "../.svn" ]]; then
```
[the remark this answers is not in the document]
Comment (nwinkler) [reply test/fixtures/svn/working/svn:6]: Thanks - I always mix these up!

Fixed in #1619.

**Pull request merged**